src/errorMessages.cpp

```diff
@@ -36,13 +36,13 @@ void set_ignored_warning(const std::vector<Warning>& warnings, bool ignore) {
 
 void printError(Warning warning, const std::string& msg) {
     static int error = 0;
-    if (MORPHIO_RAISE_WARNINGS)
-        throw MorphioError(msg);
 
     if (readers::ErrorMessages::isIgnored(warning) || MORPHIO_MAX_N_WARNINGS == 0)
         return;
 
     if (MORPHIO_MAX_N_WARNINGS < 0 || error <= MORPHIO_MAX_N_WARNINGS) {
+        if (MORPHIO_RAISE_WARNINGS)
+            throw MorphioError(msg);
```

Comment:
I think you should put it outside the :
`if(MORPHIO_MAX_N_WARNINGS < 0 || error <= MORPHIO_MAX_N_WARNINGS)` 
scope.

The result is similar but more readable imao